src/json_schema_validation.rs

```diff
@@ -0,0 +1,205 @@
+use anyhow::{bail, Context, Result};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum SchemaType {
+    String,
+    Number,
+    Integer,
+    Boolean,
+    Array,
+    Object,
+}
+
+/// Schema Validator is a JSON Schema descriptor used to evaluate the return value of a JsonPath
+/// expression, used by the presentation definition constraints field to ensure the property value
+/// meets the expected schema.
+///
+/// For more information, see the field constraints filter property:
+///
+/// https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-descriptor-object
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct SchemaValidator {
+    #[serde(rename = "type")]
+    schema_type: SchemaType,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    min_length: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    max_length: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pattern: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    minimum: Option<f64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    maximum: Option<f64>,
+    #[serde(skip_serializing_if = "Vec::is_empty", default)]
+    required: Vec<String>,
+    #[serde(skip_serializing_if = "HashMap::is_empty", default)]
+    properties: HashMap<String, Box<SchemaValidator>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    items: Option<Box<SchemaValidator>>,
+}
+
+impl PartialEq for SchemaValidator {
+    fn eq(&self, other: &Self) -> bool {
+        self.schema_type == other.schema_type
+            && self.min_length == other.min_length
+            && self.max_length == other.max_length
+            && self.pattern == other.pattern
+            && self.minimum == other.minimum
+            && self.maximum == other.maximum
+            && self.required == other.required
+            && self.properties == other.properties
+            && self.items == other.items
+    }
+}
+
+impl Eq for SchemaValidator {}
+
+impl SchemaValidator {
```

Comment:
Could you add a constructor?

Reply:
Constructor now takes a `SchemaType` to set the type of the validation.